applications/GeoMechanicsApplication/tests/cpp_tests/test_geo_mechanics_application.cpp

```diff
@@ -0,0 +1,48 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Anne van de Graaf
+//
+
+#include "testing/testing.h"
+#include "geo_mechanics_application.h"
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(ThermalAnalysisVariablesExistAfterRegistration, KratosGeoMechanicsFastSuite)
+{
+    KratosGeoMechanicsApplication geo_app;
+
+    // Note that when this unit test is run through Python, the variables have already been registered. Therefore,
+    // checking whether the thermal variables don't exist prior to registration will fail. Presumably after migrating to
+    // GoogleTest, we can add those checks here as well, to test the effect of `Register()`.
+    geo_app.Register();
+
+    const auto variable_names = std::vector<std::string>{"SPECIFIC_HEAT_CAPACITY_WATER",
+                                                         "SPECIFIC_HEAT_CAPACITY_SOLID",
+                                                         "THERMAL_CONDUCTIVITY_WATER",
+                                                         "THERMAL_CONDUCTIVITY_SOLID_XX",
+                                                         "THERMAL_CONDUCTIVITY_SOLID_YY",
+                                                         "THERMAL_CONDUCTIVITY_SOLID_ZZ",
+                                                         "THERMAL_CONDUCTIVITY_SOLID_XY",
+                                                         "THERMAL_CONDUCTIVITY_SOLID_YZ",
+                                                         "THERMAL_CONDUCTIVITY_SOLID_XZ",
+                                                         "LONGITUDINAL_DISPERSIVITY",
+                                                         "TRANSVERSE_DISPERSIVITY",
+                                                         "SOLID_COMPRESSIBILITY",
+                                                         "DT_TEMPERATURE_COEFFICIENT",
+                                                         "DT_TEMPERATURE",
+                                                         "NORMAL_HEAT_FLUX"};
+    for (const auto& name : variable_names) {
```

Comment:
```suggestion
    for (const auto& name : {"SPECIFIC_HEAT_CAPACITY_WATER",
                             "SPECIFIC_HEAT_CAPACITY_SOLID",
                             "THERMAL_CONDUCTIVITY_WATER",
                             "THERMAL_CONDUCTIVITY_SOLID_XX",
                             "THERMAL_CONDUCTIVITY_SOLID_YY",
                             "THERMAL_CONDUCTIVITY_SOLID_ZZ",
                             "THERMAL_CONDUCTIVITY_SOLID_XY",
                             "THERMAL_CONDUCTIVITY_SOLID_YZ",
                             "THERMAL_CONDUCTIVITY_SOLID_XZ",
                             "LONGITUDINAL_DISPERSIVITY",
                             "TRANSVERSE_DISPERSIVITY",
                             "SOLID_COMPRESSIBILITY",
                             "DT_TEMPERATURE_COEFFICIENT",
                             "DT_TEMPERATURE",
                             "NORMAL_HEAT_FLUX"}) {
```

[No need](https://godbolt.org/z/44ceosPjs) for the tmp vector.

Reply:
Thanks for your suggestion. We have discussed it internally and came to the conclusion that we don't see a significant benefit in adopting the suggestion. Fair enough, as you pointed out, it would eliminate a temporary variable. However, the price that we pay for that is that the `for` loop spans much more lines than before, which makes it slightly more difficult to read. Therefore, we decided to keep it the way it is.